pennylane/ops/op_math/sum.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the Sum class which contains logic for
+computing the sum of operations.
+"""
+import warnings
+import numpy as np
+import pennylane as qml
+from pennylane import math
+from pennylane.operation import Operator, expand_matrix, MatrixUndefinedError
+
+
+def sum(*summands):
+    """Compute the sum of the provided terms"""
+    return Sum(*summands)  # a wire order is required when combining operators of varying sizes
+
+
+class Sum(Operator):
+    """Arithmetic operator subclass representing the sum of operators"""
+
+    _eigs = {}  # cache eigen vectors and values like in qml.Hermitian
+
+    def __init__(self, *summands, do_queue=True, id=None):
+
+        if len(summands) < 2:
```

Comment:
Is there anything that actually breaks if we have a "sum" of just one operation?

Is this a check that indicates whether or not the code will actually break with less than two ops, or does it indicate what we think intuitively makes sense?

Reply:
It indicates what we intuitively believe makes sense. I can remove the check, or replace it with one which checks that `summands` is non-empty

Reply:
We could do some sort of preprocessing, either in the `sum` function or `__new__`.  If`sum`/`__new__` received zero or one operation, it would return either an Identity or the single operation passed.